core/src/main/scala/c4s/process/Process.scala

```diff
@@ -47,6 +50,17 @@ object Process {
         error <- Sync[F].delay(errorRef.get())
       } yield ProcessResult(ExitCode(exitValue), output, error)
 
+    def toOutputStream(opt: Option[Stream[F, Byte]]): OutputStream => F[Unit] = out =>
+      opt.fold(Sync[F].unit){ stream =>
+        Resource.fromAutoCloseableBlocking(blocker)(Sync[F].delay {out})
+          .use { outStream =>
+            stream
+              .through(writeOutputStream[F](Concurrent[F].delay(outStream), blocker))
```

Comment:
You need to disable `closeAfterUse` so the `outStream` is not closed after doing the write operation on it. That was causing the exceptions.

```suggestion
              .through(writeOutputStream[F](Concurrent[F].delay(outStream), blocker, false))
```